src/Common/tests/System/PlatformDetection.cs

```diff
@@ -264,6 +267,44 @@ private static bool IsDistroAndVersion(string distroId, string versionId = null)
             return false;
         }
 
+        private static bool IsRedHatFamilyAndVersion(string versionId = null)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                IdVersionPair v = ParseOsReleaseFile();
+                bool isRedHat = v.Id == "rhel";
+                // RedHat includes minor version. We need to account for that when comparing
+                if ((isRedHat || v.Id == "centos") && (versionId == null || v.VersionId == versionId || (isRedHat && RedHatVersionEqual(versionId, v.VersionId))))
```

Comment:
the version check logic can be written in a more global way which can include checking only concerned parts of the version. this can be done in 2 ways:

1- either you do versionId.Split('.') which will produce an array of strings of the version parts and also do v.VersionId.Split('.'). Then try to check versionId parts against v.VersionId parts (regardless how many parts is in versionId).
2- other idea is when we are reading the distro info, we read the version string and convert it to Version struct. then we can just use numbers instead of strings all the time.

if we do one of these ideas, we don't have to distinguish between isRedHat or not as you are doing in your code.

Reply:
pending (string.split) but might get changed after fixing Eric's comment